tests/acceptance/test_upload.py

```diff
@@ -150,3 +196,50 @@ def test_upload_empty_file(self):
                          "User cannot upload a valid file")
         self.assertIn(b"select a file", response.data,
                       "cannot check empty file")
+
+    def test_upload_no_user_file(self):
+        """
+        user cannot upload a data without file
+        """
+        response = self.app.post(
+            "/login",
+            data={
+                "email": "testuser@testuser.com",
+                "password": "testuser"},
+        )
+        self.assertEqual(response.status_code, 302,
+                         "Unable to login for the test user")
+
+        data = {'strategy_name': 'strategy'}  # no user_file fields
+        response = self.app.post(
+            "/upload",
+            data=data,
+        )
+
+        self.assertIn(b"No user_file is specified", response.data,
+                      "cannot check no user_file fiels")
+
+    def test_upload_empty_strategy_name(self):
+        """
+        strategy name cannot be empty
+        """
+        response = self.app.post(
```

Comment:
same issue.

If we want to use any test account, be sure to first add it first. Otherwise, how can we be 100% sure it will exist?

Reply:
@jialiaz3 Michael can create an account in the test_baseclass.py

Reply:
@jialiaz3 can you confirm if this is completed?